src/components/landing/PastAnnouncements.tsx

```diff
@@ -20,6 +21,20 @@ export default function PastAnnouncements({ announcements }: Props) {
     filterPastAnnouncements(announcements)
   );
 
+  const [selectedAnnouncement, setSelectedAnnouncement] =
+    useState<Announcement | null>(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
```

Comment:
As mentioned earlier, you can just use the value of `selectedAnnouncement` and check for nullity instead of this variable.